remotes/docker/httpreadseeker.go

```diff
@@ -77,6 +77,17 @@ func (hrs *httpReadSeeker) Read(p []byte) (n int, err error) {
 			return n, nil
 		}
 	}
+	// The CRI's imagePullProgressTimeout relies on responseBody.Close to
+	// update the process monitor's status. If the err is io.EOF, close
+	// the connection since there is no more available data.
+	if err == io.EOF {
```

Comment:
feels like it's better to be a `else if` here?

Reply:
Updated. PTAL